client/src/components/map/Map.tsx

```diff
@@ -1,188 +1,492 @@
+
+    mapRef.current = new mapboxgl.Map({
+      container: mapContainerRef.current,
+      style: "mapbox://styles/mapbox/satellite-streets-v12",
+      center: center,
+      zoom: props.zoom || defaultZoom,
+      pitch: 40,
+      interactive: true,
+    });
+
+    mapRef.current.addControl(new mapboxgl.ScaleControl(), "bottom-right");
+    mapRef.current.addControl(new mapboxgl.NavigationControl(), "bottom-right");
+    mapRef.current.addControl(new mapboxgl.FullscreenControl(), "bottom-right");
+    mapRef.current.addControl(DashboardMapDraw, "bottom-right");
+
+    mapRef.current.on("load", function () {
+      DashboardMapDraw.changeMode("static");
+    });
+
+    if (props.drawing) {
+      DashboardMapDraw.set(props.drawing);
+    }
+  }, [mapContainerRef.current]);
+
+  useEffect(() => {
+    if (props.drawing) {
+      mapRef.current?.remove();
+      mapRef.current = null;
+
+      const modes = MapboxDraw.modes;
+      modes.static = StaticMode;
+
+      mapRef.current = new mapboxgl.Map({
+        container: mapContainerRef.current,
+        style: "mapbox://styles/mapbox/satellite-streets-v12",
+        center: center,
+        zoom: props.zoom || defaultZoom,
+        pitch: 40,
+        interactive: true,
+      });
+
+      mapRef.current.addControl(new mapboxgl.ScaleControl(), "bottom-right");
+      mapRef.current.addControl(
+        new mapboxgl.NavigationControl(),
+        "bottom-right"
+      );
+      mapRef.current.addControl(
+        new mapboxgl.FullscreenControl(),
+        "bottom-right"
+      );
+      mapRef.current.addControl(DashboardMapDraw, "bottom-right");
+
+      mapRef.current.on("load", function () {
+        DashboardMapDraw.changeMode("static");
+      });
+
+      if (props.drawing) {
+        DashboardMapDraw.set(props.drawing);
+      }
+
+      // Load custom icon for points
+      mapRef.current.on("load", () => {
+        mapRef.current?.loadImage(
+          `${window.location.origin}/pin.png`,
+          (error, image: any) => {
+            if (error) throw error;
+            if (!mapRef.current?.hasImage("pin-icon")) {
+              mapRef.current?.addImage("pin-icon", image);
+            }
+
+            // Add points to map from props.drawing
+            mapRef.current?.addSource("documents", {
+              type: "geojson",
+              data: props.drawing,
             });
-            mapRef.current.addControl(PreviewMapDraw, 'bottom-right');
-            if (props.drawing) PreviewMapDraw.set(props.drawing);
-        }
-    }, [props.drawing]);
 
+            mapRef.current?.addLayer({
+              id: "document-points",
+              type: "symbol",
+              source: "documents",
+              layout: {
+                "icon-image": "pin-icon",
+                "icon-size": 1,
+              },
+            });
+          }
+        );
 
-    useEffect(() => {
-        if (!mapRef.current) return;
+        // Click event to show popup with document title
```

Comment:
showing only the title in hovering and go to document on link is more intuitive for me